tests/sentry/issues/auto_source_code_config/test_process_event.py

```diff
@@ -612,18 +629,108 @@ def test_very_short_module_name(self) -> None:
                 }
             ],
             platform=self.platform,
+            expected_in_app_stack_trace_rules=["stack.module:a.** +app"],
         )
         assert len(code_mappings) == 1
         assert code_mappings[0].stacktrace_root == "a/"
         assert code_mappings[0].source_path == "src/a/"
 
     def test_handles_dollar_sign_in_module(self) -> None:
-        # No code mapping will be stored, however, we get what would have been created
-        code_mappings, _ = self._process_and_assert_no_configuration_changes(
+        # No code mapping or in-app stack trace rule will be stored
+        code_mappings = self._process_and_assert_no_configuration_changes(
             repo_trees={REPO1: ["src/com/example/foo/Bar.kt"]},
             frames=[{"module": "com.example.foo.Bar$InnerClass", "abs_path": "Bar.kt"}],
             platform=self.platform,
+            # Notice that the in-app stack trace rule does not include "foo" as the stacktrace root does
+            expected_in_app_stack_trace_rules=["stack.module:com.example.** +app"],
         )
         assert len(code_mappings) == 1
         assert code_mappings[0].stacktrace_root == "com/example/foo/"
         assert code_mappings[0].source_path == "src/com/example/foo/"
+
+    def test_multiple_configuration_changes(self) -> None:
+        # Test case with multiple frames from different packages
+        code_mappings = self._process_and_assert_no_configuration_changes(
+            repo_trees={
+                REPO1: [
+                    "src/com/example/foo/Bar.kt",
+                    "src/com/example/foo/Baz.kt",
+                    "src/com/example/utils/Helper.kt",
+                    "src/org/other/service/Service.kt",
+                ]
+            },
+            frames=[
+                {"module": "com.example.foo.Bar", "abs_path": "Bar.kt"},
+                {"module": "com.example.foo.Baz", "abs_path": "Baz.kt"},
+                {"module": "com.example.utils.Helper", "abs_path": "Helper.kt"},
```

Comment:
There's the `utils` app and the `foo` app under the same domain, thus, we will have one in-app stack trace rule but two code mappings.